aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -68,7 +69,99 @@
 #include <numeric>
 #include <vector>
 
-namespace at { namespace native {
+namespace at {
+namespace meta {
+
+native::SCATTER_GATHER_OP get_operator_enum(const c10::string_view reduce) {
+  if (reduce == "add") {
+    return native::SCATTER_GATHER_OP::REDUCE_ADD;
+  } else if (reduce == "multiply") {
+    return native::SCATTER_GATHER_OP::REDUCE_MULTIPLY;
+  } else {
+    TORCH_CHECK(false, "reduce argument must be either add or multiply.");
+  }
+}
+
+template <typename Meta>
+void scatter_meta_impl(
+    Meta& meta,
+    const Tensor& self,
+    int64_t dim,
+    const Tensor& index,
+    const c10::optional<Tensor>& src = nullopt,
+    const c10::optional<c10::string_view> reduce = nullopt) {
+  int64_t wrapped_dim = at::maybe_wrap_dim(dim, self.dim());
+  at::native::scatter_gather_dtype_check("scatter", self, index, src);
+  at::native::scatter_shape_check(self, wrapped_dim, index, src);
+
+  meta.set_output(self.sizes(), self.options());
+
+  auto output = meta.maybe_get_output(0);
+  at::assert_no_internal_overlap(output);
+  at::assert_no_overlap(output, index);
+
+  if (src.has_value()) {
+    at::assert_no_overlap(output, src.value());
```

Comment:
This is a preexisting condition (no action needed), but huh, didn't realize overlap wasn't OK on src lol